### PR TITLE
add support for `--target_os=ios`

### PR DESCRIPTION
**Pull request by bridiver**
fixes brave/brave-browser#3080 with https://github.com/brave/brave-core/pull/2552
## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [x] Requested a security/privacy review as needed.
- [x] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions.

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: lib/config.js [hunk 1]
```diff
@@ -170,7 +191,9 @@ Config.prototype.buildArgs = function () {
 Config.prototype.shouldSign = function () {
   // it doesn't make sense to sign debug builds because the restrictions on loading
   // dynamic libs prevents them from working anyway
-  return this.mac_signing_identifier !== '' && this.buildConfig === 'Release'
+  return this.mac_signing_identifier !== '' &&
```
Comment (mihaiplesa) [inline lib/config.js]: should `--skip_signing` and `mac_installer_signing_identifier` be checked here as well?

File: lib/config.js [hunk 2]
```diff
@@ -159,6 +158,28 @@ Config.prototype.buildArgs = function () {
...
   }
 
+  if (this.targetOS === 'ios') {
+    args.target_os = 'ios'
+    args.enable_dsyms = false
+    args.enable_stripping = args.enable_dsyms
+    args.use_xcode_clang = args.is_official_build
+    args.use_clang_coverage = false
+    args.is_component_build = false
+    // args.ios_deployment_target '12.0'
```
Comment (kylehickinson) [inline lib/config.js]: This is required by the iOS lib as we use iOS 12 APIs and our minimum for brave-ios is 12. Is it possible to add this arg just for `brave-core/brave-ios` or does it have to be build-wide?

File: lib/config.js (hunk 1 above)
Comment (bridiver) [reply lib/config.js]: this just skips the entire thing for ios no matter what so `--skip_signing` would be redundant. All I did here was add `this.targetOS !== 'ios'`

File: Jenkinsfile [hunk 3]
```diff
@@ -164,7 +164,7 @@ pipeline {
                                 sh """
                                     set -e
                                     rm -rf src/brave
-                                    npm run init -- --target_os=android
+                                    npm run init -- --target_os=android --target_arch=arm64
```
Comment (bridiver) [inline Jenkinsfile]: target_arch is not a valid parameter for `init`. Only target_os is required here

Comment (mihaiplesa) [reply Jenkinsfile]: thought it was from https://github.com/brave/brave-browser/blob/master/scripts/sync.js#L16, removing now

**Pull request merged**